src/core/utils.ts

```diff
@@ -47,7 +47,8 @@ export function useWorkspaceOptions<DSType extends DataSourceBase<any>>(datasour
 export function getVariableOptions<DSType extends DataSourceBase<any>>(datasource: DSType) {
   return datasource.templateSrv
     .getVariables()
-    .map((variable) => ({ label: '$' + variable.name, value: '$' + variable.name }));
+    .filter(variable => (variable as any).current.value)
```

Comment:
@mure - Here is my proposed solution to not showing the newly-created variable in the drop-down. It's hacky, but seems to work fairly well in my testing. Let me know what you think.

Reply:
Looks like query variables have a `datasource` property:

<img width="699" alt="image" src="https://github.com/ni/systemlink-grafana-plugins/assets/9257800/0ef159e8-da06-4df4-bf33-4689c83ea042">

Maybe we could hide variables that are from the same data source the user is currently creating a variable for?

I'm also fine with not doing anything. It's not really a big deal that users see an extra autocomplete option.

Reply:
I've gone with checking the data source, it is cleaner